File: server/models/LearningLog.js
const mongoose = require('mongoose');

const learningLogSchema = new mongoose.Schema({
  user_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  stage_id: {
    type: Number,
    ref: 'GameStage',
    default: null,
  },
  idiom_id: {
    type: Number,
    ref: 'Idiom',
    required: true,
  },
  action_type: {
    type: String,
    enum: ['ATTACK', 'DEFEND', 'LEARN'],
    required: true,
  },
  chosen_difficulty: {
    type: String,
    enum: ['EASY', 'MEDIUM', 'HARD'],
    default: null,
  },
  is_correct: {
    type: Boolean,
    required: true,
  },
  response_time_ms: {
    type: Number,
    required: true,
  },
  calculated_damage: {
    type: Number,
    default: 0,
  },
  timestamp: {
    type: Date,
    default: Date.now,
  },
});

// Indexes for efficient querying
learningLogSchema.index({ user_id: 1, timestamp: -1 });
learningLogSchema.index({ user_id: 1, stage_id: 1 });
learningLogSchema.index({ user_id: 1, action_type: 1 });

module.exports = mongoose.model('LearningLog', learningLogSchema);

File: Backend/Models/LearningLog.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace IdiomLearningAPI.Models
{
    /// <summary>
    /// 학습 로그 모델 (특허 핵심 데이터)
    /// FR 4.5: 학습 성과 데이터 수신
    /// FR 6.1: 학습 데이터 수집
    /// </summary>
    [Table("LearningLogs")]
    public class LearningLog
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        public int? StageId { get; set; }

        [Required]
        public int IdiomId { get; set; }

        [Required]
        public ActionType ActionType { get; set; }

        public Difficulty? ChosenDifficulty { get; set; }

        public bool IsCorrect { get; set; }

        public int ResponseTimeMs { get; set; }

        public int CalculatedDamage { get; set; }

        public DateTime Timestamp { get; set; } = DateTime.UtcNow;
    }

    public enum ActionType
    {
        ATTACK,
        DEFEND,
        LEARN
    }
}
